script.js

```diff
@@ -87,29 +88,47 @@ function drawSnake(snakeBlocks) {
 
 function die() {
   dead = true;
-  let button = createButton("Restart");
   console.log("no");
   let dieTxt = 'Bruh';
+  dieButton();
   fill(220, 20, 60);
   textSize(30);
   text(dieTxt, 255, 280);
+  frameRate(0);
+}
+
+function dieButton(){
+  let button = createButton("Restart");
   button.id = 'restart';
   button.size(200, 100);
   button.position(200, 320);
   button.style("font-family", "Bodoni");
   button.style("font-size", "48px");
   button.mouseClicked(restartGame);
-  frameRate(0);
 }
 
 function restartGame() {
-  var b = document.getElementsByTagName("button");
+  let b = document.getElementsByTagName("button");
   b[jankyCounter].style.display = "none";
   setup();
   jankyCounter++;
 }
 
 function touchingSelf() {
+ let newPose = newPos(direction, snakeBlocks);
+  if (checkBody(snakeBlocks, newPose)) {
+    die();
+  }
+}
+
+function touchingEdge(direction, snakeBlocks) {
```

Comment:
As with `touchingSelf` I would expect this function to return a boolean and let the caller take care of the `die()`.